jetty-documentation/src/main/asciidoc/operations-guide/websocket/chapter.adoc

```diff
@@ -0,0 +1,105 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+[[og-websocket]]
+=== WebSocket
+
+WebSocket is a protocol for bidirectional communications initiated via HTTP/1.1 upgrade which provides basic message framing, layered over TCP.
+It is based on a low-level framing protocol that delivers messages in either UTF-8 TEXT or BINARY format.
+
+A single message in WebSocket can be of any size (the underlying framing however has a maximum size limit of 2^63 bytes).
+There can be an unlimited number of messages sent.
+Messages are sent sequentially, the base protocol does not support interleaved messages.
+
+A WebSocket connection goes through some basic state changes:
+
+.WebSocket connection states
+[width="50%",cols=",",options="header",]
+|=======================================================================
+|State |Description
+|CONNECTING |A HTTP Upgrade to WebSocket is in progress.
+|OPEN |The HTTP Upgrade succeeded and the socket is now open and ready to read / write.
+|CLOSING |A WebSocket Close Handshake has been started.
+|CLOSED |WebSocket is now closed, no more read/write possible.
+|=======================================================================
+
+When a WebSocket is closed, a link:{JDURL}/org/eclipse/jetty/websocket/api/StatusCode.html[status code] and short reason string is provided.
+
+[[og-ws-intro-provides]]
+==== What Jetty Provides
+
+Jetty provides an implementation of the following standards and specs.
+
+http://tools.ietf.org/html/rfc6455[RFC-6455] - The WebSocket Protocol::
++
+We support the version 13 of the released and final spec. +
+Jetty tests its WebSocket protocol implementation using the https://github.com/crossbario/autobahn-testsuite[autobahn testsuite].
+
+____
+[IMPORTANT]
```

Comment:
Admonitions should follow the correct format. See examples in Jetty 10 documentation, should be:
```
[ADMONITION]
====
Text
====
```